### ISSUE TITLE
Have OCIO build export a proper OpenColorIOConfig.cmake

### ISSUE THREAD

**Issue by lgritz** [AcademySoftwareFoundation/OpenColorIO#1187]
The modern CMake way is that packages should produce and distribute a PackageConfig.cmake, eliminating the need for all the other CMake-based packages that depend on OCIO from each, separately, having to maintain some kind of FindOpenColorIO.cmake module that contains all the searching logic.

**Comment by Yash-R** [comment]
Hey, i would like to help you with this issue can you guide me ??

**Comment by hodoulp** [comment]
Hi @Yash-R,
Happy to see a volunteer to help the OpenColorIO development.

The task requires some knowledge of `C++` and `CMake` because it involves developing a CMake utility to find a C++ library in a multi-platform project. Below are some examples:
- https://github.com/OpenImageIO/oiio/blob/master/src/cmake/modules/FindOpenColorIO.cmake
- https://github.com/PixarAnimationStudios/USD/blob/release/cmake/modules/FindOpenColorIO.cmake
 
If you still have interest in the task then start the work and the community will help you.

**Comment by Yash-R** [comment]
yes I'm very much interested to work, Can you please explain me the work briefly please ? @hodoulp

**Comment by hodoulp** [comment]
The CMake utility helps other projects to discover an installed OpenColorIO library (i.e which means where are the header files, where are the binary file(s), and what is the version). In the previous post I pasted some examples. So, OpenColorIO must have a `findOpenColorIO.cmake` file and install it.
Keep in mind that the file detection is, most of the time, platform dependant (i.e OpenColorIO supports Windows, MacOS and Linux).

Before starting anything I suggest you to start reading the documentation, and to check on how to contribute to the project.

**Comment by stilllman** [comment]
> So, OpenColorIO must have a `findOpenColorIO.cmake` file and install it.

The proper solution would be to install a config-file package (`OpenColorIOConfig.cmake`), not a find-module package. Find-modules are useful for finding libraries that are _not_ CMake-friendly, if your goal is to make OpenColorIO easily consumable through CMake then you should export your targets in a config-file. A good start would be to read the documentation here: https://cmake.org/cmake/help/latest/manual/cmake-packages.7.html

For the record, OpenColorIO 1.1 was actually already installing a config-file package, but this feature was removed by this commit apparently: https://github.com/AcademySoftwareFoundation/OpenColorIO/commit/2f5fbfc80218d5c31bdc5045df7b68e0aa149b6b

**Comment by lgritz** [comment]
I noticed today that OCIO doesn't provide a modern OpenColorIOConfig.cmake file with all the targets a downstream project should need, went to add an issue, only to discover that I had already filed this same issue right here, exactly six months ago to the day. (I guess we can place a tight upper bound on the length my memory for such things.)

So I reiterate the original request, it's what most modern cmake-based projects provide for their downstream clients, and expect from their upstream cmake-aware dependencies. I can point you to code in the cmake build systems of OIIO or OSL as examples. It's only a handful of lines to do it properly.
